docs/CppInteroperability/Roadmap.md

```diff
@@ -0,0 +1,141 @@
+
+As per our goal’s specification, this method of importing reference types allows C++ interoperability to have a clear, native mapping for a common C++ API pattern that builds a more general purpose solution off of the pre-established reference type bridging from Objective-C. Additionally, Swift preserves the same safety properties as C++ while providing an even safer option for the common pattern where reference types have a retain and release operation.
+
+### Value types
+
+As mentioned above, swift types fall into two categories: value types and reference types. Value types are relatively simple in Swift, C, and Objective-C, however, the same is not true for C++ where value types may have complex lifetime operations that are used to express non-native concepts (to Swift) such as value types that own memory. This case of owned value types is specifically difficult to map on to Swift as it is novel to the language and does not always work well with the existing model for Swift value types that are trivially copyable.  
+
+#### Trivial types
+
+This roadmap will refer to trivial value types that do not hold pointers as “trivial types.” These types include primitive types such as integers and types which are composed of other trivial types. Trivial types are “owned” types that provide trivial lifetime operations: a copy is a copy of their bits and a destroy is a no-op. Trivial types have roughly the same mapping throughout Swift, C, Objective-C, and C++ making them trivial to import. Trivial types, their instances, methods on trivial types, and other APIs that use trivial types are generally considered to be safe and usable.
+
+**Pointer types**
+
+This roadmap will refer to trivial value types that hold pointers as “pointer types.” These types include pointers themselves and types which are composed of any pointer types (potentially including other types as well). The pointers held by pointer types refer to memory that is *not owned* by the pointer type (making pointer types a “view” or “projection” into memory). While pointer types are very similar to trivial types with respect to their lifetime operations and the fact that they map similarly in these four language, they differ in the fact that while they themselves are not inherently unsafe, they may be used in unsafe APIs (discussed later).
+
+#### Owned types
```

Comment:
Is it possible to have an owned type that holds a pointer and doesn't have a copy constructor?

Reply:
Hmm, if I understand correctly, I don't think that is possible.  There would be a copy constructor, even if implicit (and fully inlined).

Reply:
OK, we need some better terminology here. (I will include this in the "definitions" section.) What I meant to ask is: "Is it possible to have an owned type that holds a pointer and has a trivial (implicit, compiler generated) copy constructor?"